File: spec/javascript/test_optional_id.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('OptionalId', 'src/fixed_struct.bin', function(r, OptionalId_) {
  assert.strictEqual(r._unnamed0, 80);
  assert.strictEqual(r._unnamed1, 65);
  assert.deepStrictEqual(r._unnamed2, new Uint8Array([67, 75, 45, 49, 255]));
});

File: spec/rust/test_optional_id.rs
extern crate kaitai_struct;
extern crate rust;

use kaitai_struct::KaitaiStruct;
use rust::OptionalId;

#[test]
fn test_optional_id() {
    if let Ok(r) = OptionalId::from_file("src/fixed_struct.bin") {

        assert_eq!(r._unnamed0, 80);
        assert_eq!(r._unnamed1, 65);
        assert_eq!(r._unnamed2, vec!([0x43, 0x4b, 0x2d, 0x31, 0xff]));
    }
}
